### PR TITLE
Update Python Curriculum Page

### LINKED ISSUE #198: Python Curriculum Page Updates
## Update Python Curriculum Page

The Python curriculum page PR was just merged, but there are some things that could be improved! 
Here's a list:

- [ ]  update home page design to match [mockup](https://www.figma.com/file/1DtxLV71aK4NKKIg0MpxbI/Learning-Pathways-Draft?node-id=0%3A1)
- [x] replace placeholder videos (lessons 1 and 2) 
- [ ] complete the curriculum!
- [ ] add practice problems to all lessons (some are currently missing them)
- [ ] refactor CSS for better maintainability

Feel free to add any other suggestions for improvements in the comments!

### PR DESCRIPTION
**Pull request by annamtong**
Updated design of the Python Curriculum Page (modified the html and sass pages) and cleaned it up based upon the mockup from #198. After talking to Einar, we decided to leave some aspects of the mockup design out though. The modifications included adding an image to the page, updating the colors and spacing of some elements to make the page look cleaner, as well as splitting the lesson cards up by group/category.  

Closes #198.

### CODE REVIEW

File: classes/py1.html
Comment (mattxwang) [inline classes/py1.html:104]: This is quite a bit of duplicated code with the previous section! Unfortunately, we can't really reformat the way the data is stored (as it is a collection), but we could instead create some sort of reusable component to represent this card. 

The Jekyll-based way we could do this is with an `_includes` "component", which we can then pass in information to render the card. I believe Jun has created something similar in [`py1-lesson-card-content`](https://github.com/uclaacm/teach-la-website/blob/master/_includes/py1-lesson-card-content.html), which you can check out! If you want an example of an include designed around parameters, you can check out [`blog-author-card.html`](https://github.com/uclaacm/teach-la-website/blob/master/_includes/blog-author-card.html); [here are the relevant docs on includes](https://jekyllrb.com/docs/includes/).

File: classes/py1.html
Comment (mattxwang) [inline classes/py1.html:10]: A minor piece of feedback, but I think it's a bit confusing to use our link styling if this thing isn't actually a link. If we want the underline effect (without the hover), we can implement that independently; however, it might also be worth picking a style that doesn't clash with our link styling.

File: classes/py1.html
Comment (mattxwang) [inline classes/py1.html:43]: This should probably be an `<h3>`, since it is under the `<h2>` category "fundamentals". Similar comment on the flow ones!

File: _sass/_py1.scss
Comment (mattxwang) [inline _sass/_py1.scss:99]: I believe we have a `.no-margin` class that already does this!

File: classes/py1.html
Comment (mattxwang) [inline classes/py1.html:9]: Is there a reason you removed this from being an `<h1>`? It doesn't seem like there's one on the page anymore!

**Review by mattxwang — CHANGES_REQUESTED** [review]
Hi Anna,

Thanks for all your work on this page, I love what you've done with it so far, and thank you for working closely with Einar! 

I've left a few comments, mostly around semantic HTML and code duplication; let me know if you need any clarification!

Also, one thing that doesn't really make it into a comment: currently if a lesson has a slide, video, and worksheet, the card gets too cramped and the time estimate looks a bit strange. I think it's up to you on how you want to resolve this (ex keep the same, move time to a new row, make the indicator links smaller), but do what makes the most sense to you!

File: classes/py1.html
Comment (annamtong) [reply classes/py1.html]: I tried a few different options and ended up deciding to leave out any effects here. With the added image directly below the title, I thought that it looked cleaner without anything, but am open to suggestions if you or @EinarBalan think including some sort of effect would look nicer!

File: classes/py1.html
Comment (annamtong) [reply classes/py1.html]: Was attempting to resolve a margins issue, but fixed it through using the `.no-margin` class and added the `<h1>` back!

File: classes/py1.html
Comment (annamtong) [reply classes/py1.html]: Thanks so much! I was having trouble figuring out how to consolidate the duplicated code and just tried creating an `_includes` that consists of all the duplicated code and a parameter specifying the lesson group.

**Comment by annamtong** [comment]
Made changes to the Python Curriculum Page addressing the issues of duplicated code and semantic HTML. Created a new [page](_includes/py1-lesson-card.html) under the `_includes` folder to address code duplication, fixed issues with some of the headings, and removed unnecessary link styling.

**Review by mattxwang — APPROVED** [review]
Thanks for the quick turnaround Anna! The way you did the `include` is pretty elegant, major respect 😊 

On my end, I think this is good to merge. Let's wait for one last approving review from @EinarBalan, and I think it's good to go! And make sure to use the squash and merge option. Great work Anna!

**Comment by annamtong** [comment]
Thanks so much for all your suggestions and help @mattxwang !

**Review by EinarBalan — APPROVED** [review]
Lgtm!

**Pull request merged**
